### ISSUE TITLE
'Continue LoRA training' broken due to adapter_model.[bin/safetensors] file extension mixup

### ISSUE THREAD

**Issue by araleza** [oobabooga/text-generation-webui#4963]
### Describe the bug

I regularly trained LoRAs down to a certain loss level, and then dropped the learning rate by hand, and then hit 'train' again to pick up from where the previous run left off.  There's a checkbox, "Override Existing Files", that is unticked so that this is the behavior.

But now after updating (a week or two since my last update), LoRA training always ignores the training done so far and creates a completely new LoRA model.

This seems to be because the LoRA is now in a file called 'adapter_model.safetensors', but the code that checks for an existing LoRA is looking for 'adapter_model.bin'.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

As described in bug

### Logs

```shell
-
```

### System Info

```shell
Ubuntu, running locally, 4090 RTX.
```

**Comment by FartyPants** [comment]
Is that what PEFT does by default, saving adapter as safetensors?

**Comment by araleza** [comment]
I think maybe a dependency of oobabooga webui has been updated to now safe as safetensors as default.  But the oobabooga Training tab loading code can only handle .bin, so I haven't managed to get it working yet.

**Comment by araleza** [comment]
I looked at some loras I trained a few weeks ago, and they do have adapter_model.bin files, which is no longer true for newly trained loras.

**Comment by araleza** [comment]
I've found that I can get it to go back to .bin files by searching for 'save_pretrained' in modules/training.py, and adding 'safe_serialization = False' as an additional optional parameter to the three calls to that function.

I don't understand why I have to do that though, because if I press F12 to 'go to definition' of save_pretrained() in the headers, VS Code shows me:
```
    def save_pretrained(
        self,
        save_directory: str,
        safe_serialization: bool = False,
        selected_adapters: Optional[list[str]] = None,
        **kwargs: Any,
    ):
        raise NotImplementedError(f"Saving is not supported for {self.__class__.__name__} (yet).")
```
The safe_serialization parameter already defaults to false there, but I'm not familiar enough with Python to understand exactly what's going on.  The error message suggests to me that this function call is overridden somewhere else, which maybe does define safe_serialization as true?

Either way, I have the .bin files back, so that should be it working until someone that knows what they're doing sorts this out properly.

**Comment by FartyPants** [comment]
Well, PEFT has now safe_serialization = True by default, that's why

![image](https://github.com/oobabooga/text-generation-webui/assets/23346289/3cf74948-acc4-46e9-b7d2-1d9334097054)

**Comment by FartyPants** [comment]
Anyhow I patched Training PRO and put the bin as default but ultimately the rest of the code should be made to work with safetensors - I just don't have time right now to mess to much with it. Once I do the changes I can transfer them to main, unless somebody does it before me - which I would prefer.

**Comment by SrVill** [comment]
When training, there is another not exactly a mistake, but a nuisance. The console constantly writes a warning from checkpoint.py (line 429):
```
"torch.utils.checkpoint: please pass in use_reentrant=True or "
            "use_reentrant=False explicitly. The default value of use_reentrant "
            "will be updated to be False in the future. To maintain current "
            "behavior, pass use_reentrant=True. It is recommended that you use "
            "use_reentrant=False. Refer to docs for more details on the "
            "differences between the two variants."
```
 Well, I turned off use_reentrant = False in line 437, but the warnings remained. How do I get rid of the warnings? I use Training_Pro

**Comment by hhjin** [comment]
I have a LoRA of safetensors trained over 3 days, to continue training on it to save the 3 days consumption, I change /modules/traning.py about line 547 to support load safetensors LoRAs:
```python
from safetensors.torch import load_file

        logger.info("Creating LoRA model")
        lora_model = get_peft_model(shared.model, config)

        if not always_override and Path(f"{lora_file_path}/adapter_model.bin").is_file():
            logger.info("Loading existing LoRA data")
            state_dict_peft = torch.load(f"{lora_file_path}/adapter_model.bin", weights_only=True)
            set_peft_model_state_dict(lora_model, state_dict_peft)
       # new code added to  load safetensors LoRAs:
        elif not always_override and Path(f"{lora_file_path}/adapter_model.safetensors").is_file():
            logger.info("Loading existing safetensors LoRA data")
            state_dict_peft = load_file(f"{lora_file_path}/adapter_model.safetensors")
            set_peft_model_state_dict(lora_model, state_dict_peft)
```
I hope this change can be applied to the main source. Thanks.  @oobabooga

**Comment by oobabooga** [comment]
The training code has been rewritten. Checkpoint resumption now uses HF Trainer checkpoint directories, and there's also a legacy fallback that handles both `.safetensors` and `.bin` adapter files. See the [release notes](https://github.com/oobabooga/text-generation-webui/releases/tag/v4.0) and [training docs](https://github.com/oobabooga/text-generation-webui/wiki/05-%E2%80%90-Training-Tab).
